### DISCUSSION TITLE
How should I get `topics` from `Log`?

### DISCUSSION THREAD

**Discussion by carlosfaria94** [wevm/viem#981] [Question] [answered]
I want to use `decodeEventLog` to decode logs that exist in a given transaction receipt. However, I cannot get `topics` from a [`Log` ](https://github.com/wagmi-dev/viem/blob/main/src/types/log.ts#L15-L44)in viem, thus I cannot use `decodeEventLog` because it's required to pass `topic`.

What I'm missing here?

**Comment by jxom** [comment, marked as answer]
`topics` are [indeed on the `Log` type](https://github.com/wagmi-dev/viem/blob/main/src/types/log.ts#L112). Just defined elsewhere. Can you post an example where you are not retrieving topics?

**Reply by carlosfaria94** [reply]
My fault! I've created a type similar to `Log` but without the topic type logic.

Thank you for point out! And thanks for this amazing work with viem!

**Reply by kyriediculous** [reply]
The `Log` type on `TransactionReceipt` is not correctly inferring these fields actually.

**Comment by antazoey** [comment]
I am getting this error in typescript:

```
index.ts:188:26 - error TS2339: Property 'topics' does not exist on type '{ address: `0x${string}`; blockHash: `0x${string}`; blockNumber: bigint; data: `0x${string}`; logIndex: number; transactionHash: `0x${string}`; transactionIndex: number; removed: boolean; }'.
```

**Reply by jxom** [reply]
Mind providing an example to reproduce?

**Reply by antazoey** [reply]
I have this: https://github.com/ApeWorX/ApePay/blob/main/sdk/js/index.ts#L188-L191
I am a bit new to Typescript and frantically trying to figure stuff out but I may have time to try and get a smaller repro some day.
